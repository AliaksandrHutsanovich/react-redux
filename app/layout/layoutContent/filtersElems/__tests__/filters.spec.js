import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Checkbox, Input } from 'antd';
import Filters from '../filters';

const mockStore = configureStore();
const initialState = {};

// jest.mock('../filters', () => ({
//   onChange: (e, dispatch) => dispatch(),
//   onChangePattern: (e, dispatch) => dispatch(),
// }));

describe('Unit test of filters', () => {
  const dispatch = jest.fn();
  const store = mockStore(initialState);
  store.dispatch = dispatch;
  const Component = shallow(<Filters store={store} searchKey="" showDone />);
  it('full render test', () => {
    expect(Component.shallow()).toMatchSnapshot();
  });

  it('checkbox should be clickable', () => {
    Component.shallow().find(Checkbox).simulate(
      'change',
      {
        target: {
          checked: true,
        },
      },
    );
    expect(dispatch).toHaveBeenCalled();
  });

  it('input text should be changable', () => {
    Component.shallow().find(Input).simulate(
      'change',
      {
        target: {
          value: 'javascript',
        },
      },
    );
    expect(dispatch).toHaveBeenCalled();
  });
});
