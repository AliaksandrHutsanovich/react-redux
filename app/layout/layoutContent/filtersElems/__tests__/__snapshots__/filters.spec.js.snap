// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test of filters full render test 1`] = `
<div
  className="filters"
>
  <Checkbox
    defaultChecked={true}
    indeterminate={false}
    onChange={[Function]}
  >
    Show done
  </Checkbox>
  <Input
    defaultValue=""
    disabled={false}
    onChange={[Function]}
    placeholder="search"
    style={
      Object {
        "width": 200,
      }
    }
    type="text"
  />
</div>
`;
